alf/algorithms/agent.py

```diff
@@ -391,36 +381,30 @@ def after_train_iter(self, experience, train_info: AgentInfo = None):
             self._goal_generator, self._reward_weight_algorithm
         ]
         algorithms = list(filter(lambda a: a is not None, algorithms))
-        self._agent_helper.after_train_iter(algorithms, experience, train_info)
+        self._agent_helper.after_train_iter(algorithms, experience, info)
 
         if self._reward_weight_algorithm:
             self._rl_algorithm.set_reward_weights(
                 self._reward_weight_algorithm.reward_weights)
 
-    def preprocess_experience(self, exp: Experience):
+    def preprocess_experience(self, root_inputs, rollout_info, batch_info):
         """Add intrinsic rewards to extrinsic rewards if there is an intrinsic
         reward module. Also call ``preprocess_experience()`` of the rl
         algorithm.
         """
-        reward = self.calc_training_reward(exp.reward, exp.rollout_info)
-        exp = exp._replace(reward=reward)
+        exp = root_inputs
+        rewards = rollout_info.rewards
+        if rewards is not ():
+            exp = exp._replace(reward=rewards['overall'])
```

Comment:
This is not correct when using normalized reward, and if we want to generate intrinsic rewards using the up-to-date goal_generator or irm instead of the version during rollout for off-policy training.